### PR TITLE
CRM-18426: added unit test and fixed notices

### PR DESCRIPTION
**Pull request by monishdeb**
- [CRM-18426: Pay later receipts not going out](https://issues.civicrm.org/jira/browse/CRM-18426)

### CODE REVIEW

**Comment by monishdeb** [comment]
@eileenmcnaughton can you verify this patch which contains -
- Unit test for ensuring mail is sent for paylater
- Notice fix where earlier for paylater workflow $result['payment_status_id'] is empty reason why $contribution->payment_status_id is not been set and eventually throw notice error "Notice: Undefined property: CRM_Contribute_BAO_Contribution::$payment_status_id in CRM_Contribute_BAO_Contribution_Utils::processConfirm() (line 180 of ..." 
- added missing pay_later_receipt template variable

File: CRM/Contribute/BAO/ContributionPage.php [hunk 1]
```diff
@@ -386,6 +386,10 @@ public static function sendMail($contactID, $values, $isTest = FALSE, $returnMes
         'receipt_date' => !$values['receipt_date'] ? NULL : date('YmdHis', strtotime($values['receipt_date'])),
       );
 
+      if (!empty($values['is_pay_later'])) {
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/ContributionPage.php]: The issue here is that we have an unreliable value in 'is_pay_later' - if you check this PR https://github.com/civicrm/civicrm-core/pull/8159/files - it is addressing the unreliability - ie. the fact that is_pay_later is being overwritten based on the contribution page value.

In general the sendMail function suffers from leakage - so I prefer to be aggressive about setting values - ie without the !empty where possible. But, these are only quick notes - I'm going to look properly later

**Comment by monishdeb** [comment]
@eileenmcnaughton  I have updated the PR

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by eileenmcnaughton** [comment]
sorry for the slow response, yeah I agree this is better than not having it - I had just intended to get my head around it a bit more

**Pull request merged**